### ISSUE TITLE
Consider supporting object notation for CSS overrides

### ISSUE THREAD

**Issue by lehni** [iocave/customize-ui#18]
This could be nice, e.g.:

```json
    "customizeUI.stylesheet": {
        ".statusbar .octicon-smiley, .statusbar .octicon-paintcan": {
          "display": "none"
        }
    },
```

**Comment by knopp** [comment]
I like the idea. Pull requests are welcome :)

**Comment by quaj** [comment]
https://github.com/iocave/customize-ui/blob/master/modules/fonts.js#L1-L17
Can i include a external library to here?
I thinks i can do this feat ^

**Comment by lehni** [comment]
I don't think you need an external library for this? Something along these lines should do?

```js
const css = typeof value === 'string' ? value : Object.entries(value).map(([key, value]) => `${key}:${value};`).join(';');
```

**Comment by knopp** [comment]
This definitely doesn't need third any third party dependency. The question is, whether it should support react like notation, i.e. backgroundColor instead of background-color, in which case it might need bit more work.

**Comment by lehni** [comment]
In that case, you can throw something like this in the mix then:

```js
function hyphenate(str) {
  return str.replace(
      /([a-z\d])([A-Z])|(\S)(?:\s+)(\S)/g,
      (all, lower, upper, beforeSpace, afterSpace) => upper
        ? `${lower}-${upper}`
        : `${beforeSpace}-${afterSpace}`
    ).toLowerCase()
}
```

**Comment by shellscape** [comment]
It's been a few years, but it would reduce maintenance burden for extensive custom css solutions if we could contain all css in a file that is loaded. There's another extension (https://marketplace.visualstudio.com/items?itemName=be5invis.vscode-custom-css) that allows loading css from file. @knopp would you be open to a contribution for that?

I'm working on a comprehensive reskin of vscode to look like Atom and I'm already up to 150 lines of CSS in settings.json. This sure would come in handy.
